src/geode/geosciences/representation/builder/structural_model_builder.cpp

```diff
@@ -43,6 +43,30 @@ namespace geode
     {
     }
 
+    void StructuralModelBuilder::copy( const StructuralModel& structural_model )
+    {
+        auto mapping = copy_components( structural_model );
+        copy_collections( mapping, structural_model );
+    }
+
+    void StructuralModelBuilder::copy_collections(
+        ComponentMapping& mapping, const StructuralModel& structural_model )
+    {
+        const auto faults_mapping = copy_faults( structural_model );
+        for( const auto& map : faults_mapping.get() )
+        {
+            mapping.collections.map( map.first, map.second );
+            register_component(
+                structural_model_.fault( map.second ).component_id() );
+            for( const auto& item :
+                structural_model.items( structural_model.fault( map.first ) ) )
+            {
+                add_item_in_collection(
+                    mapping.surfaces.in2out( item.id() ), map.second );
+            }
```

Comment:
This can be moved to RelationshipBuilder. Maybe we can find a way to move all block between l.57 and l.67 into RelationshipsBuilder. It implies I think to give as parameter a (templated?) model. But it maybe useful to copy collections from one model type to another model type.